### PR TITLE
Bring audit to monorepo

### PR DESCRIPTION
**Pull request by kobergj**
Signed-off-by: jkoberg <jkoberg@owncloud.com>

## Description

Audit service may now live in shiny mono repo

## Motivation and Context

Less maintenance when in monorepo

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: audit/pkg/config/config.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+	Events   Events   `ocisConfig:"events"`
+	Auditlog Auditlog `ocisConfig:"auditlog"`
+
+	Context context.Context
+}
+
+// Events combines the configuration options for the event bus.
+type Events struct {
+	Endpoint      string `ocisConfig:"events_endpoint" env:"AUDIT_EVENTS_ENDPOINT"`
```
Comment (wkloucek) [inline audit/pkg/config/config.go:26]: could you please add description tags? see eg. https://github.com/owncloud/ocis/blob/0dcf9524f06cf1aa639710581ec44a81065e81ab/accounts/pkg/config/config.go#L28

File: audit/pkg/config/debug.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package config
+
+// Debug defines the available debug configuration.
+type Debug struct {
+	Addr   string `ocisConfig:"addr" env:"NOTIFICATIONS_DEBUG_ADDR"`
```
Comment (wkloucek) [inline audit/pkg/config/debug.go:5]: should be `env:"AUDIT_...."`, needs also be change in some more config structs

File: ocis/pkg/runtime/service/service.go [hunk 3]
```diff
@@ -93,6 +95,7 @@ func NewService(options ...Option) (*Service, error) {
 		cfg:          opts.Config,
 	}
 
+	s.ServicesRegistry["audit"] = audit.NewSutureService
```
Comment (wkloucek) [inline ocis/pkg/runtime/service/service.go:98]: do we really wan to start it as a default service with the fullstack server? I'm not sure if it provides that much value on a raspberry pi

**Review by wkloucek — CHANGES_REQUESTED** [review]

File: ocis/pkg/runtime/service/service.go (hunk 3 above)
Comment (kobergj) [reply ocis/pkg/runtime/service/service.go:98]: Uh. You are right. And it is also written in the Jira-Ticket:
```
the audit service is default off (not part of the fullstack server start command)
```
Looks like I should practice reading. Thanks for pointing it out.

File: audit/pkg/config/config.go (hunk 1 above)
Comment (kobergj) [reply audit/pkg/config/config.go]: done

File: audit/pkg/config/debug.go (hunk 2 above)
Comment (kobergj) [reply audit/pkg/config/debug.go]: copy and paste :) sorry, fixed now

File: ocis/pkg/runtime/service/service.go (hunk 3 above)
Comment (kobergj) [reply ocis/pkg/runtime/service/service.go]: removed it

File: audit/pkg/command/health.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+// Health is the entrypoint for the health command.
+func Health(cfg *config.Config) *cli.Command {
+	return &cli.Command{
+		Name:  "health",
+		Usage: "Check health status",
+		Action: func(c *cli.Context) error {
+			// Not implemented
+			return nil
```
Comment (micbar) [inline audit/pkg/command/health.go:15]: Can we 1) either give a basic health status back or  2) create a ticket to implement it in a separate task

File: audit/pkg/command/version.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+// Version prints the service versions of all running instances.
+func Version(cfg *config.Config) *cli.Command {
+	return &cli.Command{
+		Name:     "version",
+		Usage:    "print the version of this binary and the running extension instances",
+		Category: "info",
+		Action: func(c *cli.Context) error {
+			// not implemented
```
Comment (micbar) [inline audit/pkg/command/version.go:15]: same here

File: audit/pkg/config/config.go [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+// Events combines the configuration options for the event bus.
+type Events struct {
+	Endpoint      string `ocisConfig:"events_endpoint" env:"AUDIT_EVENTS_ENDPOINT" desc:"the address of the streaming service"`
+	Cluster       string `ocisConfig:"events_cluster" env:"AUDIT_EVENTS_CLUSTER" desc:"the clusterID of the streaming service. Mandatory when using nats"`
+	ConsumerGroup string `ocisConfig:"events_group" env:"AUDIT_EVENTS_GROUP" desc:"the customergroup of the service. One group will only get one vopy of an event"`
+}
+
+// Auditlog holds audit log information
+type Auditlog struct {
+	LogToConsole bool   `ocisConfig:"log_to_console" env:"AUDIT_LOG_TO_CONSOLE" desc:"logs to Stdout if true"`
```
Comment (micbar) [inline audit/pkg/config/config.go:33]: Are these two Options exclusive or can both be set to true?

File: audit/pkg/config/config.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+	LogToFile    bool   `ocisConfig:"log_to_file" env:"AUDIT_LOG_TO_FILE" desc:"logs to file if true"`
+	FilePath     string `ocisConfig:"filepath" env:"AUDIT_FILEPATH" desc:"filepath to the logfile. Mandatory if LogToFile is true"`
+	Format       string `ocisConfig:"format" env:"AUDIT_FORMAT" desc:"log format. using json is advised"`
```
Comment (micbar) [inline audit/pkg/config/config.go:36]: Which other options besides json are possible?

**Review by micbar — APPROVED** [review]
Some minor questions.

File: audit/pkg/config/config.go (hunk 6 above)
Comment (kobergj) [reply audit/pkg/config/config.go:33]: both can be set to true

File: audit/pkg/config/config.go (hunk 7 above)
Comment (kobergj) [reply audit/pkg/config/config.go:36]: At the moment none. But I figured it be beneficial to have it configurable. I can remove and hardcode it if you say thats too confusing.

**Review by wkloucek — APPROVED** [review]

**Pull request merged**